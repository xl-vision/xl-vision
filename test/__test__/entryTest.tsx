/* eslint-disable global-require */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-unsafe-argument */

describe('Entry', () => {
  it('@xl-vision/react', () => {
    const components = require('../../packages/react/src');
    expect(Object.keys(components)).toMatchSnapshot();
  });
  it('@xl-vision/icons', () => {
    const components = require('../../packages/icons/src');
    expect(Object.keys(components)).toMatchSnapshot();
  });
});
