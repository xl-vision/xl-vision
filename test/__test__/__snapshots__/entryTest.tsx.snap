// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entry @xl-vision/hooks 1`] = `
Array [
  "useLatestRef",
  "useConstantFn",
  "useForkRef",
  "useIsomorphicLayoutEffect",
  "useMedia",
  "usePrevious",
  "useMount",
  "useUnmount",
  "useResizeObserver",
  "useTransition",
  "useCssTransition",
]
`;

exports[`Entry @xl-vision/icons 1`] = `
Array [
  "createIcon",
  "AccountBookFilled",
  "AlertFilled",
  "AlipayCircleFilled",
  "AlipaySquareFilled",
  "AliwangwangFilled",
  "AmazonCircleFilled",
  "AmazonSquareFilled",
  "AndroidFilled",
  "ApiFilled",
  "AppleFilled",
  "AppstoreFilled",
  "AudioFilled",
  "BackwardFilled",
  "BankFilled",
  "BehanceCircleFilled",
  "BehanceSquareFilled",
  "BellFilled",
  "BookFilled",
  "BoxPlotFilled",
  "BugFilled",
  "BuildFilled",
  "BulbFilled",
  "CalculatorFilled",
  "CalendarFilled",
  "CameraFilled",
  "CarFilled",
  "CaretDownFilled",
  "CaretLeftFilled",
  "CaretRightFilled",
  "CaretUpFilled",
  "CarryOutFilled",
  "CheckCircleFilled",
  "CheckSquareFilled",
  "ChromeFilled",
  "CiCircleFilled",
  "ClockCircleFilled",
  "CloseCircleFilled",
  "CloseSquareFilled",
  "CloudFilled",
  "CodeSandboxCircleFilled",
  "CodeSandboxSquareFilled",
  "CodeFilled",
  "CodepenCircleFilled",
  "CodepenSquareFilled",
  "CompassFilled",
  "ContactsFilled",
  "ContainerFilled",
  "ControlFilled",
  "CopyFilled",
  "CopyrightCircleFilled",
  "CreditCardFilled",
  "CrownFilled",
  "CustomerServiceFilled",
  "DashboardFilled",
  "DatabaseFilled",
  "DeleteFilled",
  "DiffFilled",
  "DingtalkCircleFilled",
  "DingtalkSquareFilled",
  "DislikeFilled",
  "DollarCircleFilled",
  "DownCircleFilled",
  "DownSquareFilled",
  "DribbbleCircleFilled",
  "DribbbleSquareFilled",
  "DropboxCircleFilled",
  "DropboxSquareFilled",
  "EditFilled",
  "EnvironmentFilled",
  "EuroCircleFilled",
  "ExclamationCircleFilled",
  "ExperimentFilled",
  "EyeInvisibleFilled",
  "EyeFilled",
  "FacebookFilled",
  "FastBackwardFilled",
  "FastForwardFilled",
  "FileAddFilled",
  "FileExcelFilled",
  "FileExclamationFilled",
  "FileImageFilled",
  "FileMarkdownFilled",
  "FilePdfFilled",
  "FilePptFilled",
  "FileTextFilled",
  "FileUnknownFilled",
  "FileWordFilled",
  "FileZipFilled",
  "FileFilled",
  "FilterFilled",
  "FireFilled",
  "FlagFilled",
  "FolderAddFilled",
  "FolderOpenFilled",
  "FolderFilled",
  "FormatPainterFilled",
  "ForwardFilled",
  "FrownFilled",
  "FundFilled",
  "FunnelPlotFilled",
  "GiftFilled",
  "GithubFilled",
  "GitlabFilled",
  "GoldFilled",
  "GoldenFilled",
  "GoogleCircleFilled",
  "GooglePlusCircleFilled",
  "GooglePlusSquareFilled",
  "GoogleSquareFilled",
  "HddFilled",
  "HeartFilled",
  "HighlightFilled",
  "HomeFilled",
  "HourglassFilled",
  "Html5Filled",
  "IdcardFilled",
  "IeCircleFilled",
  "IeSquareFilled",
  "InfoCircleFilled",
  "InstagramFilled",
  "InsuranceFilled",
  "InteractionFilled",
  "LayoutFilled",
  "LeftCircleFilled",
  "LeftSquareFilled",
  "LikeFilled",
  "LinkedinFilled",
  "LockFilled",
  "MacCommandFilled",
  "MailFilled",
  "MedicineBoxFilled",
  "MediumCircleFilled",
  "MediumSquareFilled",
  "MehFilled",
  "MessageFilled",
  "MinusCircleFilled",
  "MinusSquareFilled",
  "MobileFilled",
  "MoneyCollectFilled",
  "NotificationFilled",
  "PauseCircleFilled",
  "PayCircleFilled",
  "PhoneFilled",
  "PictureFilled",
  "PieChartFilled",
  "PlayCircleFilled",
  "PlaySquareFilled",
  "PlusCircleFilled",
  "PlusSquareFilled",
  "PoundCircleFilled",
  "PrinterFilled",
  "ProfileFilled",
  "ProjectFilled",
  "PropertySafetyFilled",
  "PushpinFilled",
  "QqCircleFilled",
  "QqSquareFilled",
  "QuestionCircleFilled",
  "ReadFilled",
  "ReconciliationFilled",
  "RedEnvelopeFilled",
  "RedditCircleFilled",
  "RedditSquareFilled",
  "RestFilled",
  "RightCircleFilled",
  "RightSquareFilled",
  "RobotFilled",
  "RocketFilled",
  "SafetyCertificateFilled",
  "SaveFilled",
  "ScheduleFilled",
  "SecurityScanFilled",
  "SettingFilled",
  "ShopFilled",
  "ShoppingFilled",
  "SignalFilled",
  "SketchCircleFilled",
  "SketchSquareFilled",
  "SkinFilled",
  "SkypeFilled",
  "SlackCircleFilled",
  "SlackSquareFilled",
  "SlidersFilled",
  "SmileFilled",
  "SnippetsFilled",
  "SoundFilled",
  "StarFilled",
  "StepBackwardFilled",
  "StepForwardFilled",
  "StopFilled",
  "SwitcherFilled",
  "TabletFilled",
  "TagFilled",
  "TagsFilled",
  "TaobaoCircleFilled",
  "TaobaoSquareFilled",
  "ThunderboltFilled",
  "ToolFilled",
  "TrademarkCircleFilled",
  "TrophyFilled",
  "TwitterCircleFilled",
  "TwitterSquareFilled",
  "UnlockFilled",
  "UpCircleFilled",
  "UpSquareFilled",
  "UsbFilled",
  "VideoCameraFilled",
  "WalletFilled",
  "WarningFilled",
  "WechatFilled",
  "WeiboCircleFilled",
  "WeiboSquareFilled",
  "WindowsFilled",
  "YahooFilled",
  "YoutubeFilled",
  "YuqueFilled",
  "ZhihuCircleFilled",
  "ZhihuSquareFilled",
  "AccountBookOutlined",
  "AimOutlined",
  "AlertOutlined",
  "AlibabaOutlined",
  "AlignCenterOutlined",
  "AlignLeftOutlined",
  "AlignRightOutlined",
  "AlipayCircleOutlined",
  "AlipayOutlined",
  "AliwangwangOutlined",
  "AliyunOutlined",
  "AmazonOutlined",
  "AndroidOutlined",
  "AntCloudOutlined",
  "AntDesignOutlined",
  "ApartmentOutlined",
  "ApiOutlined",
  "AppleOutlined",
  "AppstoreAddOutlined",
  "AppstoreOutlined",
  "AreaChartOutlined",
  "ArrowDownOutlined",
  "ArrowLeftOutlined",
  "ArrowRightOutlined",
  "ArrowUpOutlined",
  "ArrowsAltOutlined",
  "AudioMutedOutlined",
  "AudioOutlined",
  "AuditOutlined",
  "BackwardOutlined",
  "BankOutlined",
  "BarChartOutlined",
  "BarcodeOutlined",
  "BarsOutlined",
  "BehanceSquareOutlined",
  "BehanceOutlined",
  "BellOutlined",
  "BgColorsOutlined",
  "BlockOutlined",
  "BoldOutlined",
  "BookOutlined",
  "BorderBottomOutlined",
  "BorderHorizontalOutlined",
  "BorderInnerOutlined",
  "BorderLeftOutlined",
  "BorderOuterOutlined",
  "BorderRightOutlined",
  "BorderTopOutlined",
  "BorderVerticleOutlined",
  "BorderOutlined",
  "BorderlessTableOutlined",
  "BoxPlotOutlined",
  "BranchesOutlined",
  "BugOutlined",
  "BuildOutlined",
  "BulbOutlined",
  "CalculatorOutlined",
  "CalendarOutlined",
  "CameraOutlined",
  "CarOutlined",
  "CaretDownOutlined",
  "CaretLeftOutlined",
  "CaretRightOutlined",
  "CaretUpOutlined",
  "CarryOutOutlined",
  "CheckCircleOutlined",
  "CheckSquareOutlined",
  "CheckOutlined",
  "ChromeOutlined",
  "CiCircleOutlined",
  "CiOutlined",
  "ClearOutlined",
  "ClockCircleOutlined",
  "CloseCircleOutlined",
  "CloseSquareOutlined",
  "CloseOutlined",
  "CloudDownloadOutlined",
  "CloudServerOutlined",
  "CloudSyncOutlined",
  "CloudUploadOutlined",
  "CloudOutlined",
  "ClusterOutlined",
  "CodeSandboxOutlined",
  "CodeOutlined",
  "CodepenCircleOutlined",
  "CodepenOutlined",
  "CoffeeOutlined",
  "ColumnHeightOutlined",
  "ColumnWidthOutlined",
  "CommentOutlined",
  "CompassOutlined",
  "CompressOutlined",
  "ConsoleSqlOutlined",
  "ContactsOutlined",
  "ContainerOutlined",
  "ControlOutlined",
  "CopyOutlined",
  "CopyrightCircleOutlined",
  "CopyrightOutlined",
  "CreditCardOutlined",
  "CrownOutlined",
  "CustomerServiceOutlined",
  "DashOutlined",
  "DashboardOutlined",
  "DatabaseOutlined",
  "DeleteColumnOutlined",
  "DeleteRowOutlined",
  "DeleteOutlined",
  "DeliveredProcedureOutlined",
  "DeploymentUnitOutlined",
  "DesktopOutlined",
  "DiffOutlined",
  "DingdingOutlined",
  "DingtalkOutlined",
  "DisconnectOutlined",
  "DislikeOutlined",
  "DollarCircleOutlined",
  "DollarOutlined",
  "DotChartOutlined",
  "DoubleLeftOutlined",
  "DoubleRightOutlined",
  "DownCircleOutlined",
  "DownSquareOutlined",
  "DownOutlined",
  "DownloadOutlined",
  "DragOutlined",
  "DribbbleSquareOutlined",
  "DribbbleOutlined",
  "DropboxOutlined",
  "EditOutlined",
  "EllipsisOutlined",
  "EnterOutlined",
  "EnvironmentOutlined",
  "EuroCircleOutlined",
  "EuroOutlined",
  "ExceptionOutlined",
  "ExclamationCircleOutlined",
  "ExclamationOutlined",
  "ExpandAltOutlined",
  "ExpandOutlined",
  "ExperimentOutlined",
  "ExportOutlined",
  "EyeInvisibleOutlined",
  "EyeOutlined",
  "FacebookOutlined",
  "FallOutlined",
  "FastBackwardOutlined",
  "FastForwardOutlined",
  "FieldBinaryOutlined",
  "FieldNumberOutlined",
  "FieldStringOutlined",
  "FieldTimeOutlined",
  "FileAddOutlined",
  "FileDoneOutlined",
  "FileExcelOutlined",
  "FileExclamationOutlined",
  "FileGifOutlined",
  "FileImageOutlined",
  "FileJpgOutlined",
  "FileMarkdownOutlined",
  "FilePdfOutlined",
  "FilePptOutlined",
  "FileProtectOutlined",
  "FileSearchOutlined",
  "FileSyncOutlined",
  "FileTextOutlined",
  "FileUnknownOutlined",
  "FileWordOutlined",
  "FileZipOutlined",
  "FileOutlined",
  "FilterOutlined",
  "FireOutlined",
  "FlagOutlined",
  "FolderAddOutlined",
  "FolderOpenOutlined",
  "FolderViewOutlined",
  "FolderOutlined",
  "FontColorsOutlined",
  "FontSizeOutlined",
  "ForkOutlined",
  "FormOutlined",
  "FormatPainterOutlined",
  "ForwardOutlined",
  "FrownOutlined",
  "FullscreenExitOutlined",
  "FullscreenOutlined",
  "FunctionOutlined",
  "FundProjectionScreenOutlined",
  "FundViewOutlined",
  "FundOutlined",
  "FunnelPlotOutlined",
  "GatewayOutlined",
  "GifOutlined",
  "GiftOutlined",
  "GithubOutlined",
  "GitlabOutlined",
  "GlobalOutlined",
  "GoldOutlined",
  "GooglePlusOutlined",
  "GoogleOutlined",
  "GroupOutlined",
  "HddOutlined",
  "HeartOutlined",
  "HeatMapOutlined",
  "HighlightOutlined",
  "HistoryOutlined",
  "HomeOutlined",
  "HourglassOutlined",
  "Html5Outlined",
  "IdcardOutlined",
  "IeOutlined",
  "ImportOutlined",
  "InboxOutlined",
  "InfoCircleOutlined",
  "InfoOutlined",
  "InsertRowAboveOutlined",
  "InsertRowBelowOutlined",
  "InsertRowLeftOutlined",
  "InsertRowRightOutlined",
  "InstagramOutlined",
  "InsuranceOutlined",
  "InteractionOutlined",
  "IssuesCloseOutlined",
  "ItalicOutlined",
  "KeyOutlined",
  "LaptopOutlined",
  "LayoutOutlined",
  "LeftCircleOutlined",
  "LeftSquareOutlined",
  "LeftOutlined",
  "LikeOutlined",
  "LineChartOutlined",
  "LineHeightOutlined",
  "LineOutlined",
  "LinkOutlined",
  "LinkedinOutlined",
  "Loading3QuartersOutlined",
  "LoadingOutlined",
  "LockOutlined",
  "LoginOutlined",
  "LogoutOutlined",
  "MacCommandOutlined",
  "MailOutlined",
  "ManOutlined",
  "MedicineBoxOutlined",
  "MediumWorkmarkOutlined",
  "MediumOutlined",
  "MehOutlined",
  "MenuFoldOutlined",
  "MenuUnfoldOutlined",
  "MenuOutlined",
  "MergeCellsOutlined",
  "MessageOutlined",
  "MinusCircleOutlined",
  "MinusSquareOutlined",
  "MinusOutlined",
  "MobileOutlined",
  "MoneyCollectOutlined",
  "MonitorOutlined",
  "MoreOutlined",
  "NodeCollapseOutlined",
  "NodeExpandOutlined",
  "NodeIndexOutlined",
  "NotificationOutlined",
  "NumberOutlined",
  "OneToOneOutlined",
  "OrderedListOutlined",
  "PaperClipOutlined",
  "PartitionOutlined",
  "PauseCircleOutlined",
  "PauseOutlined",
  "PayCircleOutlined",
  "PercentageOutlined",
  "PhoneOutlined",
  "PicCenterOutlined",
  "PicLeftOutlined",
  "PicRightOutlined",
  "PictureOutlined",
  "PieChartOutlined",
  "PlayCircleOutlined",
  "PlaySquareOutlined",
  "PlusCircleOutlined",
  "PlusSquareOutlined",
  "PlusOutlined",
  "PoundCircleOutlined",
  "PoundOutlined",
  "PoweroffOutlined",
  "PrinterOutlined",
  "ProfileOutlined",
  "ProjectOutlined",
  "PropertySafetyOutlined",
  "PullRequestOutlined",
  "PushpinOutlined",
  "QqOutlined",
  "QrcodeOutlined",
  "QuestionCircleOutlined",
  "QuestionOutlined",
  "RadarChartOutlined",
  "RadiusBottomleftOutlined",
  "RadiusBottomrightOutlined",
  "RadiusSettingOutlined",
  "RadiusUpleftOutlined",
  "RadiusUprightOutlined",
  "ReadOutlined",
  "ReconciliationOutlined",
  "RedEnvelopeOutlined",
  "RedditOutlined",
  "RedoOutlined",
  "ReloadOutlined",
  "RestOutlined",
  "RetweetOutlined",
  "RightCircleOutlined",
  "RightSquareOutlined",
  "RightOutlined",
  "RiseOutlined",
  "RobotOutlined",
  "RocketOutlined",
  "RollbackOutlined",
  "RotateLeftOutlined",
  "RotateRightOutlined",
  "SafetyCertificateOutlined",
  "SafetyOutlined",
  "SaveOutlined",
  "ScanOutlined",
  "ScheduleOutlined",
  "ScissorOutlined",
  "SearchOutlined",
  "SecurityScanOutlined",
  "SelectOutlined",
  "SendOutlined",
  "SettingOutlined",
  "ShakeOutlined",
  "ShareAltOutlined",
  "ShopOutlined",
  "ShoppingCartOutlined",
  "ShoppingOutlined",
  "ShrinkOutlined",
  "SisternodeOutlined",
  "SketchOutlined",
  "SkinOutlined",
  "SkypeOutlined",
  "SlackSquareOutlined",
  "SlackOutlined",
  "SlidersOutlined",
  "SmallDashOutlined",
  "SmileOutlined",
  "SnippetsOutlined",
  "SolutionOutlined",
  "SortAscendingOutlined",
  "SortDescendingOutlined",
  "SoundOutlined",
  "SplitCellsOutlined",
  "StarOutlined",
  "StepBackwardOutlined",
  "StepForwardOutlined",
  "StockOutlined",
  "StopOutlined",
  "StrikethroughOutlined",
  "SubnodeOutlined",
  "SwapLeftOutlined",
  "SwapRightOutlined",
  "SwapOutlined",
  "SwitcherOutlined",
  "SyncOutlined",
  "TableOutlined",
  "TabletOutlined",
  "TagOutlined",
  "TagsOutlined",
  "TaobaoCircleOutlined",
  "TaobaoOutlined",
  "TeamOutlined",
  "ThunderboltOutlined",
  "ToTopOutlined",
  "ToolOutlined",
  "TrademarkCircleOutlined",
  "TrademarkOutlined",
  "TransactionOutlined",
  "TranslationOutlined",
  "TrophyOutlined",
  "TwitterOutlined",
  "UnderlineOutlined",
  "UndoOutlined",
  "UngroupOutlined",
  "UnlockOutlined",
  "UnorderedListOutlined",
  "UpCircleOutlined",
  "UpSquareOutlined",
  "UpOutlined",
  "UploadOutlined",
  "UsbOutlined",
  "UserAddOutlined",
  "UserDeleteOutlined",
  "UserSwitchOutlined",
  "UserOutlined",
  "UsergroupAddOutlined",
  "UsergroupDeleteOutlined",
  "VerifiedOutlined",
  "VerticalAlignBottomOutlined",
  "VerticalAlignMiddleOutlined",
  "VerticalAlignTopOutlined",
  "VerticalLeftOutlined",
  "VerticalRightOutlined",
  "VideoCameraAddOutlined",
  "VideoCameraOutlined",
  "WalletOutlined",
  "WarningOutlined",
  "WechatOutlined",
  "WeiboCircleOutlined",
  "WeiboSquareOutlined",
  "WeiboOutlined",
  "WhatsAppOutlined",
  "WifiOutlined",
  "WindowsOutlined",
  "WomanOutlined",
  "YahooOutlined",
  "YoutubeOutlined",
  "YuqueOutlined",
  "ZhihuOutlined",
  "ZoomInOutlined",
  "ZoomOutOutlined",
]
`;

exports[`Entry @xl-vision/react 1`] = `
Array [
  "styled",
  "createGlobalStyles",
  "ThemeProvider",
  "ThemeContext",
  "useTheme",
  "ConfigProvider",
  "ConfigContext",
  "useConfig",
  "defaultConfigContext",
  "Transition",
  "CollapseTransition",
  "TransitionGroup",
  "Row",
  "Ripple",
  "Button",
  "BaseButton",
  "Portal",
  "CssBaseline",
  "Popper",
  "Tooltip",
  "Popover",
  "Popconfirm",
  "Dropdown",
  "Modal",
  "Dialog",
  "Avatar",
  "Input",
  "TextArea",
  "Affix",
  "ResizeObserver",
  "BackTop",
  "Anchor",
]
`;

exports[`Entry @xl-vision/styled-engine 1`] = `
Array [
  "createGlobalStyles",
  "keyframes",
  "StyledEngineProvider",
  "ThemeContext",
  "styled",
]
`;

exports[`Entry @xl-vision/utils 1`] = `
Array [
  "warning",
  "noop",
  "raf",
  "nextFrame",
  "getComputedStyle",
  "getDocumentElement",
  "getBoundingClientRect",
  "deepMerge",
  "onTransitionEnd",
  "getTransitionInfo",
  "addClass",
  "getClasses",
  "removeClass",
  "containClass",
  "on",
  "off",
  "setRef",
  "EventEmitter",
  "isBrowser",
  "isServer",
  "isProduction",
  "isDevelopment",
  "isNode",
  "isHTMLElement",
  "isElement",
  "isShadowRoot",
  "isDocument",
  "oneOf",
  "isFirefox",
  "isWindow",
  "getWindow",
  "contains",
]
`;
